packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -1176,6 +1195,18 @@ export const DatePickerMixin = (subclass) =>
           this._ignoreFocusedDateChange = false;
         }
       }
+
+      this.__setEnteredDate(this._inputElementValue);
+    }
+
+    /** @private */
+    __setEnteredDate(value) {
+      const parsedDate = this.__parseDate(value);
+      if (value && !dateEquals(parsedDate, this.__enteredDate)) {
+        this.__enteredDate = parsedDate;
+      } else if (this.__enteredDate != null) {
+        this.__enteredDate = null;
```

Comment:
The `this.__enteredDate != null` check prevents reassigning the property if it hasn't changed. Setting a sync property to the same value can still trigger a sync update if there have been other updates scheduled already.

Reply:
This sounds like a problem in the PolylitMixin to me. Could you please create a separate issue?
IMO we should check whether property has changed like Lit does [internally](https://github.com/lit/lit/blob/37c025029a085a4837496b6a4fa7203034c3d16f/packages/reactive-element/src/reactive-element.ts#L1258-L1260).

Reply:
Done: https://github.com/vaadin/web-components/issues/8221